src/GraphQL/Types/Scalars/DateTimeOffsetGraphType.cs

```diff
@@ -20,14 +20,13 @@ public DateTimeOffsetGraphType()
         }
 
         /// <inheritdoc/>
-        public override object ParseLiteral(IValue value) => value switch
-        {
-            DateTimeOffsetValue offsetValue => offsetValue.Value,
-            StringValue stringValue => ParseValue(stringValue.Value),
-            _ => null
-        };
+        public override object ParseLiteral(IValue value)
+            => value is StringValue stringValue ? ParseValue(stringValue.Value) : null;
 
         /// <inheritdoc/>
         public override object ParseValue(object value) => ValueConverter.ConvertTo(value, typeof(DateTimeOffset));
+
+        /// <inheritdoc/>
+        public override IValue ToAST(object value) => new StringValue(((DateTimeOffset)value).ToString("O"));
```

Comment:
What is the reason for "O" format specifier ? Is this code migrated from somewhere?

Reply:
We should match it to the default serialization format of the `System.Text.Json` serializer for this type.  I'm not sure if that is `O` or not.  I chose `O` because it is a roundtrip format and looked most like what I've seen in the past.  It could be wrong.

Reply:
😄  `AstPrinter.Print(new DateTimeOffsetValue(DateTimeOffset.Now));` yields "" (empty string)

The same for `DateTimeValue` and or course for the rest of removed values because they all are not registered in `AstPrintVisitor`.

Now they will be handled as `StringValue`. So... any format seems OK.

Reply:
Added comment for format specifier.  It is correct as it matches the description of the the scalar (specifying ISO 8601), and matches the default format of System.Text.Json (also ISO 8601).

Reply:
So we are good with AstPrinter?  That was just a comment?

Reply:
Just a comment. `AstPrinter` refactoring is out of scope for v4.